app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -531,41 +531,35 @@ private static void recordSuccessfulSyncTime(String username) {
         //this is the temporary implementation of everything past this point
 
         //Wipe storage
-        //TODO: move table instead. Should be straightforward with sandboxed db's
-        wipeStorageForFourTwelveSync();
+        SQLiteDatabase userDb = CommCareApplication.instance().getUserDbHandle();
+        wipeStorageForFourTwelveSync(userDb);
 
         String failureReason = "";
         try {
-            //Get new data
-            String syncToken = readInput(cache.retrieveCache(), factory);
+            String syncToken = readInputWithoutCommit(cache.retrieveCache(), factory);
             updateUserSyncToken(syncToken);
             Logger.log(LogTypes.TYPE_USER, "Sync Recovery Successful");
+            userDb.setTransactionSuccessful();
             return new Pair<>(PROGRESS_DONE, "");
-        } catch (ActionableInvalidStructureException e) {
-            e.printStackTrace();
-            failureReason = e.getLocalizedMessage();
         } catch (InvalidStructureException | XmlPullParserException
-                | UnfullfilledRequirementsException | SessionUnavailableException | IOException e) {
-            e.printStackTrace();
-            failureReason = e.getMessage();
+                | UnfullfilledRequirementsException | SessionUnavailableException
+                | IOException e) {
+            Logger.exception("Sync recovery failed|" + e.getLocalizedMessage(), e);
         } finally {
             //destroy temp file

```

Comment:
can you move this comment to be back above the `cache.release()` line, since that's what it describes?

Reply:
Nice catch, thanks